### PR TITLE
Add cores= and memory= keywords to scale

### PR DESCRIPTION
**Pull request by mrocklin**
Dask-Jobqueue did this internally.
It seems like a decent idea to pull upstream.

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @guillaumeeb for review (I think that you did this originally?)

File: distributed/deploy/spec.py [hunk 1]
```diff
@@ -330,7 +331,31 @@ def __exit__(self, typ, value, traceback):
...
+                raise ValueError(
+                    "to use scale(memory=...) your worker definition must include a memory_limit definition"
+                )
+
+        if cores is not None:
+            try:
+                n = max(n, int(math.ceil(cores / self.new_spec["options"]["nthreads"])))
+            except KeyError:
+                raise ValueError(
+                    "to use scale(threads=...) your worker definition must include an nthreads= definition"
```
Comment (jrbourbeau) [inline distributed/deploy/spec.py]: Minor comment

```suggestion
                    "to use scale(cores=...) your worker definition must include an nthreads= definition"
```

**Comment by guillaumeeb** [comment]
Currently testing Dask-jobqueue SpecCluster PR. This does not work because dask-jobqueue uses the custom Job class, which does not need memory or nthreads argument, or at least not in every implementations. For PBS, `cluster.new_spec` looks like this :

```
{'cls': dask_jobqueue.pbs.PBSJob,
 'options': {'cores': 4,
  'memory': '20Gb',
  'walltime': '01:00:00',
  'interface': None,
  'protocol': 'tcp://',
  'security': None}}
```

Should we modify this in dask-jobqueue's side?

We also lack the nice widget for filling cores or memory in jupyter, see #2985

**Comment by mrocklin** [comment]
I believe I have resolved this latest comment.  We now check for a variety of terms, like cores, threads, nthreads, and so on

**Comment by mrocklin** [comment]
Although the widget issue is still live.  Thank you for raising an issue.

**Comment by guillaumeeb** [comment]
Will try to test again all this next week, but not sure I can find the time. 

Thanks a lot for all the work!

**Pull request merged**